src/bal-converter/index.ts

```diff
@@ -0,0 +1,178 @@
+    label: Object.entries(labels).map(([isoCode, value]) => ({
+      isoCode,
+      value,
+    })), // TODO: rename key to 'labels'
+    type: { value: "voie" }, // TODO: How to get the type from the BAL?
+    geometry: {
+      type: "Point",
+      coordinates: [balAdresse.long, balAdresse.lat],
+    },
+    dateMAJ: balAdresse.date_der_maj,
+  };
+};
+
+const balToBan = (bal: Bal): Ban => {
+  const ban = bal.reduce(
+    (acc: Ban, balAdresse: BalAdresse) => {
+      const { banID, banCommonTopoID } = getIds(balAdresse);
+      const banIdContent = balAddrToBanAddr(balAdresse, acc.addresses?.[banID]);
+      const banCommonTopoIdContent = balTopoToBanTopo(
+        balAdresse,
+        acc.commonToponyms?.[banCommonTopoID]
+      );
+      const districtID = banIdContent.codeCommune;
+      return {
+        ...acc,
+        districtID,
+        addresses: {
+          ...acc.addresses,
+          [banID]: banIdContent,
+        },
+        commonToponyms: {
+          ...acc.commonToponyms,
+          [banCommonTopoID]: banCommonTopoIdContent,
+        },
+      };
+    },
+    {
+      districtID: "",
+      // districts: {},
+      addresses: {},
+      commonToponyms: {},
+    }
+  );
+
+  return ban;
+};
+
+export const sendBalToBan = async (bal: string) => {
+  const balJSON = csvToJson(bal) as Bal;
+  const { districtID, addresses, commonToponyms } = balToBan(balJSON);
+
+  const banAddressIds: BanID[] = Object.keys(addresses || {});
+  const banAddresses: BanAddresses = Object.values(addresses || {});
+  const addressIdsReport = await getAddressIdsReport(districtID, banAddressIds);
+
+  const banToponymIds: BanCommonTopoID[] = Object.keys(commonToponyms || {});
+  const banToponyms: BanCommonToponyms = Object.values(commonToponyms || {});
+  const toponymsIdsReport = await getCommonToponymIdsReport(
+    districtID,
+    banToponymIds
+  );
+
+  // Sort Addresses (Add/Update/Delete)
+  const banAddressesToAdd = banAddresses.filter(({ id }) =>
+    addressIdsReport.idsToCreate.includes(id)
+  );
+  const banAddressesToUpdate = banAddresses.filter(({ id }) =>
+    addressIdsReport.idsToUpdate.includes(id)
+  );
+  const banAddressesIdsToDelete = banAddressIds.filter((id) =>
+    addressIdsReport.idsToDelete.includes(id)
+  );
+
+  // Sort Toponyms (Add/Update/Delete)
+  const banToponymsToAdd = banToponyms.filter(({ id }) =>
+    toponymsIdsReport.idsToCreate.includes(id)
+  );
+  const banToponymsToUpdate = banToponyms.filter(({ id }) =>
+    toponymsIdsReport.idsToUpdate.includes(id)
+  );
+  const banToponymsIdsToDelete = banToponymIds.filter((id) =>
+    toponymsIdsReport.idsToDelete.includes(id)
+  );
+
+  return Promise.all([
+    // Bulk Actions to BAN?
+    createAddresses(banAddressesToAdd),
+    updateAddresses(banAddressesToUpdate),
+    deleteAddresses(banAddressesIdsToDelete),
+    createCommonToponyms(banToponymsToAdd),
+    updateCommonToponyms(banToponymsToUpdate),
+    deleteCommonToponyms(banToponymsIdsToDelete),
```

Comment:
```suggestion
    banAddressesToAdd.length > 0 && createAddresses(banAddressesToAdd),
    banAddressesToUpdate.length > 0 && updateAddresses(banAddressesToUpdate),
    banAddressesIdsToDelete.length > 0 && deleteAddresses(banAddressesIdsToDelete),
    banToponymsToAdd.length > 0 && createCommonToponyms(banToponymsToAdd),
    banToponymsToUpdate.length > 0 && updateCommonToponyms(banToponymsToUpdate),
    banToponymsIdsToDelete.length > 0 && deleteCommonToponyms(banToponymsIdsToDelete),
```

Il faudrait peut-être conditionner les call vers ban-plateforme si seulement les adresses/toponymes ne sont pas vides.